test/integration/advanced.js

```diff
@@ -51,11 +51,65 @@ describe('bitcoinjs-lib (advanced)', function () {
           if (err) return done(err)
 
           var actual = bitcoin.Transaction.fromHex(transaction.txHex)
-          var dataScript2 = actual.outs[0].script
-          var data2 = bitcoin.script.decompile(dataScript2)[1]
+          var actualScript = actual.outs[0].script
+          assert.deepEqual(actualScript, dataScript)
 
-          assert.deepEqual(dataScript, dataScript2)
-          assert.deepEqual(data, data2)
+          done()
+        })
+      })
+    })
+  })
+
+  it('can create a transaction using OP_HODL/OP_CHECKLOCKTIMEVERIFY', function (done) {
+    this.timeout(30000)
+
+    var network = bitcoin.networks.testnet
+    var keyPair = bitcoin.ECPair.makeRandom({ network: network })
+    var address = keyPair.getAddress()
+
+    blockchain.t.faucet(address, 2e4, function (err, unspents) {
+      if (err) return done(err)
+
+      // use the oldest unspent
+      var unspent = unspents.pop()
+      var tx = new bitcoin.TransactionBuilder(network)
+
+      // now + 1 month
+      var hodlDate = Math.floor((Date.now() + new Date(0).setMonth(1)) / 1000)
+      var hodlLockTimeBuffer = new Buffer(4)
+      hodlLockTimeBuffer.writeInt32LE(hodlDate | 0, 0)
+
+      // {signature} {signature} or
+      // OP_0 {signature} after 1 month
+      var hodlScript = bitcoin.script.compile([
```

Comment:
Its an arbitrary example, sure, 1 signature or 2 signatures, but, it works, and its clear how to extend it IMHO.

Until we have something better? :)